lxd/storage_zfs.go

```diff
@@ -1836,6 +1854,12 @@ func (s *storageZfs) zfsPoolCreate() error {
 		return err
 	}
 
+	fixperms = shared.VarPath("storage-pools", s.pool.Name, "deleted")
+	err = os.Chmod(fixperms, deletedDirMode)
+	if err != nil {
+		logger.Warnf("failed to chmod \"%s\" to \"0%s\": %s", fixperms, strconv.FormatInt(int64(deletedDirMode), 8), err)
+	}
```

Comment:
delete/* should always have mountpoint unset so there is no reason for "deleted" to ever exist on the filesystem.